platform/admin/admin-configuration-configinstaller/src/main/java/org/codice/ddf/admin/configuration/ConfigurationInstaller.java

```diff
@@ -13,204 +13,259 @@
+      throw new IllegalStateException(msg);
     }
+
+    // Write to disk if necessary (4)
+    writeIfNecessary(context, cachedConfigData);
   }
 
   /**
-   * Remove config files from etc when a delete occurs but only if felix is tracking the file.
+   * {@inheritDoc}
+   *
+   * <p>Remove config files from etc when a delete occurs but only if felix is tracking the file.
    *
-   * @param event the configuration event for which a delete is needed
+   * @param pid the pid pointing to the configuration to delete.
    */
-  private void handleDelete(ConfigurationEvent event) {
-    String pid = event.getPid();
-    File felixFileOriginal = pidFileMap.remove(pid);
-    if (felixFileOriginal != null) {
+  @Override
+  public void handleDelete(String pid) throws IOException {
+    CachedConfigData cachedConfigData = pidDataMap.remove(pid);
+    File fileFromCache = (cachedConfigData != null) ? cachedConfigData.getFelixFile() : null;
+    if (fileFromCache != null && fileFromCache.exists()) {
       LOGGER.debug("Deleting file because config was deleted for pid {}", pid);
       try {
-        performOperation(configActions.delete(felixFileOriginal.toPath()));
-      } catch (IllegalArgumentException | ConfiguratorException e) {
+        Files.delete(fileFromCache.toPath());
+        SECURITY_LOGGER.info("Removing a deleted config: {}", fileFromCache.toPath());
+      } catch (IOException e) {
         LOGGER.debug("Problem deleting config file: ", e);
+        // Synchronous with config admin, so we can report the failure to the UI this way
+        throw e;
       }
     }
   }
 
-  private void performOperation(Operation operation) {
-    Configurator configurator = configuratorFactory.getConfigurator();
-    configurator.add(operation);
-    OperationReport report = configurator.commit("Synchronizing the etc directory");
-    if (report.containsFailedResults()) {
-      LOGGER.error("One or more file operations failed, see debug logs for more info");
-      if (LOGGER.isDebugEnabled()) {
-        report
-            .getFailedResults()
-            .stream()
-            .map(Result::getError)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .map(Exception.class::cast)
-            .forEach(e -> LOGGER.debug("Failed file operation: ", e));
+  /**
+   * In this block, based upon the circumstances of where it gets called, there's a guarantee that
+   * {@link ConfigurationContext#getConfigFile()} will not return null.
+   */
+  private void writeIfNecessary(ConfigurationContext context, CachedConfigData data)
+      throws IOException {
+    File dest = context.getConfigFile();
+    if (dest.exists()) {
+      // Improvement: These would become checksums
+      Dictionary<String, Object> fileState = data.getProps();
+      Dictionary<String, Object> configState = context.getSanitizedProperties();
+      processPasswords(context.getServicePid(), configState, encryptionService::encryptValue);
```

Comment:
❗️ I know you asked if documentation should be added. I think it definitely should because, if I'm understanding correctly, these newly encrypted values will be in-memory and will show up in the AdminUI. We wouldn't want to surprise the admin's with this change.